### PR TITLE
Solve #51

### PR DESCRIPTION
**Pull request by Wicpar**
- Solves #51
- added `merge `merge and `merge_with` to `PathItem`
- `ApiRouter` now merges `PathItem` properly

### CODE REVIEW

**Comment by netthier** [comment]
I just noticed, neither of your PRs follow the [conventional commits spec](https://www.conventionalcommits.org/en/v1.0.0/) as specified in CONTRIBUTING.md
You can rewrite the commit message using `git commit --amend`, just add something like `feat(aide):` in front.

**Comment by Wicpar** [comment]
I wasn't aware of the standard, i'll look into it.

File: crates/aide/Cargo.toml [hunk 1]
```diff
@@ -31,8 +31,8 @@ cfg-if = "1.0.0"
 # custom axum extractors
 serde_qs = { version = "0.12.0", optional = true }
 jwt-authorizer = { version = "0.9", optional = true }
-axum-sqlx-tx = { version = "0.5", optional = true }
-sqlx = { version = "0.6", optional = true }
+axum-sqlx-tx = { git = "https://github.com/Wicpar/axum-sqlx-tx.git", optional = true }
```
Comment (netthier) [inline crates/aide/Cargo.toml:34]: What's up with this pointing at your own custom fork, are you planning to maintain that library in the future?

Comment (Wicpar) [reply crates/aide/Cargo.toml:34]: It's obviously a mistake, i need to create a proper build using my fork and cannot wait on the library accepting the changes or not.
I'll redo a proper PR when i have the time.

**Pull request closed without merging**